File: csharp/UntangledConditionalsTest/CapturingLogger.cs
using System;
using System.Collections.Generic;
using UntangledConditionals;

namespace TestProject1
{
    internal class CapturingLogger : Logger
    {
        public List<string> Lines { get; } = new List<string>();

        public void info(string message)
        {
            Lines.Add("INFO: " + message);
        }

        public void error(string message)
        {
            Lines.Add("ERROR: " + message);
        }
    }
}

File: typescript/test/CapturingLogger.ts
import {Logger} from "../src/dependencies/Logger";

export class CapturingLogger implements Logger {
    lines: Array<string> = [];

    info(message: string): void {
        this.lines.push("INFO: " + message);
    }

    error(message: string): void {
        this.lines.push("ERROR: " + message);
    }
}
